### PR TITLE
MOD-7459 skip json tests when no json module is found

### PR DESCRIPTION
**Pull request by DvirDukhan**
**Describe the changes in the pull request**

This PR adds the ability to `skip` JSON tests when no JSON module is found
**Which issues this PR fixes**
MOD-7459

**Main objects this PR modified**
1. Added `NOJSON` option for `skip` decorator on pytest
2. Added this skip option on all JSON pytests

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: tests/pytests/common.py [hunk 1]
```diff
@@ -380,6 +380,8 @@ def wrapper():
                 raise SkipTest()
             if gc_no_fork and Env().cmd(config_cmd(), 'GET', 'GC_POLICY')[0][1] != 'fork':
                 raise SkipTest()
+            if NOJSON and not REJSON:
```
Comment (GuyAv46) [inline tests/pytests/common.py:383]: to support both `True` and `False` we should have it like we do with `cluster`:
```suggestion
            if NOJSON == REJSON:
```

File: tests/pytests/includes.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 CI = os.getenv('CI', '') != ''
 TEST_DEBUG = os.getenv('TEST_DEBUG', '0') == '1'
 MT_BUILD = os.getenv('REDISEARCH_MT_BUILD', '0') == '1'
+REJSON = os.getenv('REJSON', '0') == '1'
```
Comment (GuyAv46) [inline tests/pytests/includes.py:24]: is it 1 by default?

File: tests/pytests/test_empty.py [hunk 3]
```diff
@@ -393,7 +393,26 @@ def testEmptyTagHash(env, conn, idx, dialect):
...
+        # empty value indexing
+        env.expect('FT.CREATE', 'idx', 'SCHEMA', 't', 'TAG', 'INDEXEMPTY').ok()
+        n_docs = 1000
+        for i in range(n_docs):
+            conn.execute_command('HSET', f'h{i}', 't', '' if i % 2 == 0 else f'{i}')
+        res = env.cmd('FT.SEARCH', 'idx', '@t:{""}', 'WITHCOUNT', 'LIMIT', '0', '0')
+        env.assertEqual(int(res[0]), 500)
+        res = env.cmd('FT.SEARCH', 'idx', '-@t:{""}', 'WITHCOUNT', 'LIMIT', '0', '0')
+        env.assertEqual(int(res[0]), 500)
+        env.flush()
```
Comment (GuyAv46) [inline tests/pytests/test_empty.py:406]: Not relevant to JSON?

File: tests/pytests/test_expire.py [hunk 4]
```diff
@@ -72,27 +72,48 @@ def buildExpireDocsResults(isSortable, isJson):
...
+@skip(cluster=True, redis_less_than="7.2", NOJSON=True)
+def testExpireDocsJson(env):
 
     for isJson in [False, True]:
-        expected_results = buildExpireDocsResults(False, isJson)
-        expireDocs(env, False, # Without SORTABLE - since the fields are not SORTABLE, we need to load the results from Redis Keyspace
-                expected_results, isJson)
+        expected_results = buildExpireDocsResults(False, True)
+        # Without SORTABLE - since the fields are not SORTABLE, we need to load the results from Redis Keyspace
+        expireDocs(env, False, expected_results, True)
```
Comment (GuyAv46) [inline tests/pytests/test_expire.py:88]: No need for the loop here

File: tests/pytests/includes.py (hunk 2 above)
Comment (DvirDukhan) [reply tests/pytests/includes.py:24]: yes

**Review by GuyAv46 — APPROVED** [review]

**Pull request merged**